### PR TITLE
Issue #13809: Kill mutation in Checker2

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13809: Kill mutation in Checker2

https://github.com/checkstyle/checkstyle/blob/223ed3981ebf088c005dc7c8bf2d075194f77ae1/config/pitest-suppressions/pitest-common-suppressions.xml#L39-L46

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java [hunk 1]
```diff
@@ -314,7 +314,8 @@ private void processFiles(List<File> files) throws CheckstyleException {
                 }
 
                 // We need to catch all errors to put a reason failure (file name) in error
-                throw new Error("Error was thrown while processing " + file.getPath(), error);
+                throw new Error("Error was thrown while processing "
+                        + file.getAbsolutePath(), error);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java:318]: Can we keep getPath ?
Looks like just extra assert is required in tests

**Review by romani — CHANGES_REQUESTED** [review]
CI is red 

Items

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java (hunk 1 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/Checker.java:318]: I think yes can you please guide in which sceanrio we are throwing this error

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/Checker.java:318]: I think test is failing at https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=17141&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=1322

We have full coverage so you can easily find it by message or change type of exception and some test will fail

**Comment by romani** [comment]
@Kevin222004 , please resolve conflict

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java (hunk 1 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: restored changes

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java [hunk 2]
```diff
@@ -1706,6 +1706,49 @@ public void testViolationMessageOnIoException() throws Exception {
         verify(checkerConfig, filePath, expected);
     }
 
+    /**
+     * Test doesn't need to be serialized.
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:1710]: `This test is created to cover pitest mutation over Error message. Usually Exceptions are thrown , not Error, so we need to mock creation of Error.`

**Review by romani — CHANGES_REQUESTED** [review]
Good.
Small item to improve before merge:

**Comment by romani** [comment]
@Kevin222004 , please update this PR, looks like easy change is required.

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java (hunk 2 above)
Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Please remove file length suppression and synthetic test, we should be able to kill this mutation with code improvement in Checker. We can extract duplicate calls to `File#getPath` in `Checker#process` to a variable before `try` block.

**Comment by romani** [comment]
```✘-1 ~/java/github/romani/checkstyle [Kevin222004/Checker2 L|✚ 2⚑ 1] 
08:15 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java b/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java
index 83d2096..083adff 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java
@@ -283,40 +283,39 @@ public class Checker extends AbstractAutomaticBean implements MessageDispatcher,
     // -@cs[CyclomaticComplexity] no easy way to split this logic of processing the file
     private void processFiles(List<File> files) throws CheckstyleException {
         for (final File file : files) {
-            String fileName = null;
+            String absolutePath = file.getAbsolutePath();
             try {
-                fileName = file.getAbsolutePath();
                 final long timestamp = file.lastModified();
-                if (cacheFile != null && cacheFile.isInCache(fileName, timestamp)
-                        || !acceptFileStarted(fileName)) {
+                if (cacheFile != null && cacheFile.isInCache(absolutePath, timestamp)
+                        || !acceptFileStarted(absolutePath)) {
                     continue;
                 }
                 if (cacheFile != null) {
-                    cacheFile.put(fileName, timestamp);
+                    cacheFile.put(absolutePath, timestamp);
                 }
-                fireFileStarted(fileName);
+                fireFileStarted(absolutePath);
                 final SortedSet<Violation> fileMessages = processFile(file);
-                fireErrors(fileName, fileMessages);
-                fireFileFinished(fileName);
+                fireErrors(absolutePath, fileMessages);
+                fireFileFinished(absolutePath);
             }
             // -@cs[IllegalCatch] There is no other way to deliver filename that was under
             // processing. See https://github.com/checkstyle/checkstyle/issues/2285
             catch (Exception ex) {
-                if (fileName != null && cacheFile != null) {
-                    cacheFile.remove(fileName);
+                if (absolutePath != null && cacheFile != null) {
+                    cacheFile.remove(absolutePath);
                 }
 
                 // We need to catch all exceptions to put a reason failure (file name) in exception
                 throw new CheckstyleException("Exception was thrown while processing "
-                        + file.getPath(), ex);
+                        + absolutePath, ex);
             }
             catch (Error error) {
-                if (fileName != null && cacheFile != null) {
-                    cacheFile.remove(fileName);
+                if (absolutePath != null && cacheFile != null) {
+                    cacheFile.remove(absolutePath);
                 }
 
                 // We need to catch all errors to put a reason failure (file name) in error
-                throw new Error("Error was thrown while processing " + file.getPath(), error);
+                throw new Error("Error was thrown while processing " + absolutePath, error);
             }
         }
     }
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
index 3598117..71f2da4 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
@@ -53,6 +53,7 @@ import java.util.SortedSet;
 import java.util.TreeSet;
 import java.util.stream.Collectors;
 
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
@@ -1715,6 +1716,7 @@ public class CheckerTest extends AbstractModuleTestSupport {
      *      for test does not require serialization
      */
     @Test
+    @Disabled
     public void testCatchErrorWithFileName() throws Exception {
         final String errorMessage = "Error was thrown while processing ";
         final Error expectedError = new IOError(new InternalError(errorMessage));
```

inthis case testExceptionWithCacheAndNoFileName failing 
```
Error cause differs from IOError
value of            : throwable.getCause()
expected instance of: java.io.IOError
but was instance of : java.lang.InternalError
with value          : java.lang.InternalError: Java Virtual Machine is broken or has run out of resources necessary for it to continue operating.

	at com.puppycrawl.tools.checkstyle.CheckerTest.testCatchErrorWithCacheWithNoFileName(CheckerTest.java:1256)

```

**Comment by nrmancuso** [comment]
Changes at https://github.com/checkstyle/checkstyle/pull/13905#issuecomment-1784141080 are not what I am requesting.

**Comment by romani** [comment]
But it was strange that we operated by absolute path, but reported some path. We might try to remove test of security exception. It may might be any operation on path are you exception prone, so that is why it was in try

**Comment by nrmancuso** [comment]
See https://github.com/checkstyle/checkstyle/pull/13981

**Comment by romani** [comment]
merged in form of https://github.com/checkstyle/checkstyle/pull/13981

**Pull request closed without merging**